prem_utils/connectors/replicate.py

```diff
@@ -81,3 +88,81 @@ def chat_completion(
             "provider_id": "replicate",
         }
         return plain_response
+
+    def finetuning(self, model: str, training_data: dict, validation_data: dict | None = None, num_epochs: int = 3):
+        training_url = self._upload_and_transform_data(training_data, n_samples=25)
+        model_name = f"{model.split(':')[0].split('/')[1]}-{int(time.time())}"
+        fine_tuned_model = self.client.models.create(
+            owner="filopedraz",
+            name=model_name,
+            visibility="private",
+            hardware="cpu",
+        )
+        if validation_data:
+            validation_url = self._upload_and_transform_data(validation_data, n_samples=5)
+        else:
+            validation_url = None
+        training = self.client.trainings.create(
+            version=model,
+            input={
+                "train_data": training_url,
+                "validation_data": validation_url,
+                "num_train_epochs": num_epochs,
+                "batch_size_training": 1,
+                "val_batch_size": 1,
+            },
+            destination=f"{fine_tuned_model.owner}/{fine_tuned_model.name}",
+        )
+        return training.id
```

Comment:
Each provider is returning different data struct. We should comply with OpenAI response.